File: JS Exams/01.  Basics Exam Collection/Basics Online Exam -  28 and 29 March 2020/06. Tournament of Christmas/06. Tournament of Christmas.js
function tournamentOfChristmas(arg) {
   let days = arg.shift()
   let [winGames, lostGames, allMoney] = [0, 0, 0]
   
   while (days--) {
      let [win, lost] = [0, 0]
      while (true) {
         currentSport = arg.shift()
         if (currentSport === 'Finish') break
         arg.shift() === 'win' ? win += 20 : lost += 20
      }

      win > lost ? win *= 1.10 : win
      allMoney += win
      winGames += win
      lostGames += lost
   }

   if (winGames > lostGames)
      return `You won the tournament! Total raised money: ${(allMoney * 1.20).toFixed(2)}`
   return `You lost the tournament! Total raised money: ${allMoney.toFixed(2)}`
}


// console.log(tournamentOfChristmas([2,
//    'volleyball',
//    'win',
//    'football',
//    'lose',
//    'basketball',
//    'win',
//    'Finish',
//    'golf',
//    'win',
//    'tennis',
//    'win',
//    'badminton',
//    'win',
//    'Finish',
// ]))

// console.log(tournamentOfChristmas([
// 3,
// 'darts',
// 'lose',
// 'handball',
// 'lose',
// 'judo',
// 'win',
// 'Finish',
// 'snooker',
// 'lose',
// 'swimming',
// 'lose',
// 'squash',
// 'lose',
// 'table tennis',
// 'win',
// 'Finish',
// 'volleyball',
// 'win',
// 'basketball',
// 'win',
// 'Finish'
// ]))

File: Python Exams/01. Basic Exam Collection/Basics Online Exam -  28 and 29 March 2020/06. Tournament of Christmas/06. Tournament of Christmas.py
days = int(input())
all_money = 0
current_money = 0
current_win_games = 0
current_lost_games = 0
win_games = 0
lost_games = 0

for _ in range(0, days ):
     
   while True:
      game = input()

      if game == 'Finish':
         if current_win_games > current_lost_games:
             current_money = current_money * 1.10
             all_money += current_money
         else:    
             all_money += current_money

         current_money = 0
         current_win_games = 0
         current_lost_games = 0
         break
        
      win_or_lose = input()
       
      if  win_or_lose  == 'win':
        current_money += 20
        current_win_games += 1
        win_games += 1
      else:
         current_lost_games += 1
         lost_games += 1

if win_games > lost_games:
  print(f'You won the tournament! Total raised money: {(all_money * 1.20):.2f}')
else:  
   print(f'You lost the tournament! Total raised money: {all_money:.2f}')